dask/dataframe/io/parquet/arrow.py

```diff
@@ -265,149 +91,540 @@ def _index_in_schema(index, schema):
+    dtypes = _get_pyarrow_dtypes(schema, categories)
+    dtypes = {storage_name_mapping.get(k, k): v for k, v in dtypes.items()}
+
+    index_cols = index or ()
+    meta = _meta_from_dtypes(all_columns, dtypes, index_cols, column_index_names)
+    meta = clear_known_categories(meta, cols=categories)
+
+    if partition_obj:
+        for partition in partition_obj:
+            if isinstance(index, list) and partition.name == index[0]:
+                meta.index = pd.CategoricalIndex(
+                    categories=partition.keys, name=index[0]
+                )
+            elif partition.name == meta.index.name:
+                meta.index = pd.CategoricalIndex(
+                    categories=partition.keys, name=meta.index.name
+                )
+            elif partition.name in meta.columns:
+                meta[partition.name] = pd.Series(
+                    pd.Categorical(categories=partition.keys, values=[]),
+                    index=meta.index,
+                )
+
+    return meta, index_cols, categories, index
+
+
+def _aggregate_stats(
+    file_path, file_row_group_stats, file_row_group_column_stats, stat_col_indices
+):
+    """ Utility to aggregate the statistics for N row-groups
+        into a single dictionary.
+    """
+    if len(file_row_group_stats) < 1:
+        raise ValueError("Empty statistics.")
+    elif len(file_row_group_column_stats) == 0:
+        assert len(file_row_group_stats) == 1
+        return file_row_group_stats[0]
+    else:
+        df_rgs = pd.DataFrame(file_row_group_stats)
+        s = {
+            "file_path_0": file_path,
+            "num-rows": df_rgs["num-rows"].sum(),
+            "total_byte_size": df_rgs["total_byte_size"].sum(),
+            "columns": [],
+        }
+        df_cols = pd.DataFrame(file_row_group_column_stats)
+        for ind, name in enumerate(stat_col_indices):
+            i = ind * 3
+            s["columns"].append(
+                {
+                    "name": name,
+                    "min": df_cols.iloc[:, i].min(),
+                    "max": df_cols.iloc[:, i + 1].max(),
+                    "null_count": df_cols.iloc[:, i + 2].sum(),
+                }
             )
+        return s
 
-        column_names, index_names = _normalize_index_columns(
-            columns, column_names + partitions, index, index_names
-        )
 
-        all_columns = index_names + column_names
+def _process_metadata(
+    metadata, single_rg_parts, gather_statistics, stat_col_indices, no_filters
+):
+    # Get the number of row groups per file
+    file_row_groups = defaultdict(list)
+    file_row_group_stats = defaultdict(list)
+    file_row_group_column_stats = defaultdict(list)
+    cmax_last = {}
+    for rg in range(metadata.num_row_groups):
+        row_group = metadata.row_group(rg)
+        fpath = row_group.column(0).file_path
+        if fpath is None:
+            raise ValueError("metadata is missing file_path string.")
```

Comment:
Technically, I think the parquet standard allows for some row-groups in the current file, and some in another file, but I've never seen this happen

Reply:
Thats right -  However, this check is only for metadata collected from a _metadata file (or manually aggregated by us),  not for footer metadata of a "data" file.  Since we are relying on the path information in the global metadata file, we require the path to be specified.

Reply:
I suppose the error message could be expanded to state, that we assume a file called "_metadata" contains no data - can this be done without making it too confusing? (Not that we expect this error ever to be  hit)

Reply:
I will improve the error message here.

>(Not that we expect this error ever to be hit)

ArrowEngine used to fail to populate the file_path fields for partitioned datasets when writing the global _metadata file.  So, it is possible for this to be hit when reading an old dataset.